### PR TITLE
chore: replace library tinydir with std::filesystem

### PR DESCRIPTION
**Pull request by federico-sysdig**
**What type of PR is this?**
/kind cleanup

**Any specific area of the project related to this PR?**
/area build

**Does this PR require a change in the driver versions?**
no

**What this PR does / why we need it**:
Replaces the usage of library `tinydir`, used only once to scan a directory of Lua source files, with `std::filesystem` from the standard C++ library. The dependency from `tinydir` is also removed from the project.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
I'd like to know how I can get evidence that the modified function is behaving the same or, if not possible, if someone can provide some support for adding a test for it.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by FedeDP** [comment]
> I'd like to know how I can get evidence that the modified function is behaving the same or, if not possible, if someone can provide some support for adding a test for it.

I think we need to write unit tests for `get_chisel_list` :) 
You can create a new test folder under `userspace/libsinsp/test` and add a new test there? Of course the new folder should only be included when we build with chisels enabled.

File: userspace/chisel/chisel.cpp [hunk 1]
```diff
@@ -1074,84 +1075,43 @@ struct filename
...
-
-			filename fn = split_filename(string(file.name));
-			if(fn.ext != "sc" && fn.ext != "lua")
-			{
-				goto next_file;
-			}
-
-			for(vector<chisel_desc>::const_iterator it_desc = chisel_descs->begin();
-				it_desc != chisel_descs->end(); ++it_desc)
+			if(dir_entry.path().extension() == ".lua")
```
Comment (FedeDP) [inline userspace/chisel/chisel.cpp:1095]: We were accepting `sc` too:
```
if(fn.ext != "sc" && fn.ext != "lua")
```

Comment (FedeDP) [reply userspace/chisel/chisel.cpp:1095]: ah i see, we only inited lua chisel ending with lua, but eventually added to a vector the `sc` too (if asked). 
I don't get why though :)

**Review by FedeDP — APPROVED** [review]
This LGTM; the code is pretty small and the logic very simple, i think it's good to go.
/approve

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**